packages/fetch-router/README.md

```diff
@@ -664,6 +665,177 @@ let button = html`<button>${icon} Click me</button>` // icon is not escaped
 
 See the [`@remix-run/html-template` documentation](https://github.com/remix-run/remix/tree/main/packages/html-template#readme) for more details.
 
+### Working with Files
+
+The router provides a couple of tools for serving files:
+
+- **`file()` response helper** - The primitive for returning file responses with full HTTP semantics
+- **`staticFiles()` middleware** - Convenience middleware for serving files from a directory
+
+#### The `file()` Response Helper
+
+The `file()` response helper returns a `Response` for a file with full HTTP semantics, including:
+
+- **Content-Type** and **Content-Length** headers
+- **ETag** generation (weak or strong)
+- **Last-Modified** headers
+- **Cache-Control** headers
+- **Conditional requests** (`If-None-Match`, `If-Modified-Since`, `If-Match`, `If-Unmodified-Since`)
+- **Range requests** for partial content (`206 Partial Content`)
+- **HEAD** request support
+
+```ts
+import * as res from '@remix-run/fetch-router/response-helpers'
+import { openFile } from '@remix-run/lazy-file/fs'
+
+router.get('/assets/:filename', async (context) => {
+  let file = await openFile(`./assets/${context.params.filename}`)
+
+  return res.file(file, context)
+})
+```
+
+##### Options
+
+The `file()` helper accepts an optional third argument with configuration options:
+
+```ts
+return res.file(file, context, {
```

Comment:
Context could be Headers / SuperHeaders.

Reply:
We need the request method too, so I've opted to accept a `request` arg here rather than the full context.